### DISCUSSION TITLE
Using Third-Party Extensions?

### DISCUSSION THREAD

**Discussion by splashsky** [php/frankenphp#838] [Q&A]
Howdy! The docs are kinda-ish clear on adjusting what extensions you can bundle into the binary, but what if you have a third-party extension like [this libsql one](https://github.com/darkterminal/libsql-extension)? I'd like to test it out and use it in my project running on Franken but it's unclear how to go about doing that.

Thanks!

**Comment by dunglas** [comment]
Hi, if you compile FrankenPHP by yourself or use the Docker images, it's straightforward: you can install your custom extension as usual.

For the static binary, it's a bit more complex. First, your extension needs to be compatible with static compilation, then you have to use the extension points provided by static-php-cli (or provide a PR to static-php-cli if the extension is open source): https://static-php.dev/en/guide/manual-build.html#inject-external-script

**Reply by splashsky** [reply]
Thanks for the quick reply, and the awesome project. Is there a guide to compiling Franken with the additional extension? It compiles to `.so` if that's correct

**Reply by dunglas** [reply]
If you compiled FrankenPHP manually or used the Docker image. Just reference the `.so` in your `php.ini` as usual. There is nothing specific to FrankenPHP to do.
